File: Dart/insertionSort.dart
/*
best case time complexity: O(n) and worst cate time complexity: O(n^2)
*/
import 'Dart:io';

List<int> insertionSort({required List<int> list}) {
  for (int i = 1; i < list.length; i++) {
    int item = list[i];
    int j = i - 1;
    while (j >= 0 && list[j] > item) {
      list[j + 1] = list[j];
      j--;
    }
    list[j + 1] = item;
  }
  return list;
}

void main() {
  print("Enter index number of a list: ");
  int? indexNo = int.parse(stdin.readLineSync()!);

  List<int> list = [];
  print("Enter elements of the list:");
  for (int i = 0; i < indexNo; i++) {
    list.add(int.parse(stdin.readLineSync()!));
  }
  print("Initial list: $list");
  List<int> sortedList = insertionSort(list: list);
  print("Sorted list: $sortedList");
}

File: Python/insertionSort.py
def insertionSort(list1):
    
    for i in range(1, len(list1)):
        item = list1[i]
        j=i-1
        while j >= 0 and list1[j] > item:
            list1[j+1]=list1[j]
            j=j-1
        list1[j+1]=item
    return list1

list1 = list(map(int, input("Enter all elements of the list: ").split()))

print(f"Initial list: {list1}")
print(f"Sorted list: {insertionSort(list1=list1)}")